components/inputs/input-date.js

```diff
@@ -297,25 +333,34 @@ class InputDate extends LocalizeCoreElement(LitElement) {
 		}
 	}
 
-	_handleSetToToday() {
+	async _handleSetToToday() {
 		const date = getToday();
-		this._updateValueDispatchEvent(formatDateInISO(date));
+		await this._updateValueDispatchEvent(formatDateInISO(date));
 		this._dropdown.close();
 		this.focus();
 	}
 
 	_setFormattedValue() {
-		this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : (this.emptyText ? this.emptyText : '');
+		this._formattedValue = this._shownValue ? formatISODateInUserCalDescriptor(this._shownValue) : (this.emptyText ? this.emptyText : '');
 	}
 
-	_updateValueDispatchEvent(dateInISO) {
-		if (dateInISO === this.value) return;
+	async _updateValueDispatchEvent(dateInISO) {
+		if (dateInISO === this._shownValue) return; // prevent validation from happening multiple times for same change
+		this._shownValue = dateInISO;
+		this.validationCustomValid = !dateInISO || isDateInRange(getDateFromISODate(dateInISO), getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue));
+		await this.validate();
```

Comment:
`validate` returns a list of errors that occurred so I'd recommend checking the result of this to determine if validation succeeded.

But we do need some discussion on whether we want `validate` to be called by the components themselves or if they need to be wrapped in a form / validation group to perform automatic validation.

The main reason for having the form / validation group do it is that we need to support both native and custom form elements and native elements have no way of calling their own `validate` function. We might be able to do some hybrid but definitely something that should be discussed.

Reply:
Yup for sure - when I had talked to Jeff about how input-date should validate, he wanted the validation to happen on enter or blur, so it did seem to make the most sense to put the validation directly in the component. However, if the component would be only within this form/validation group that would do the automatic validation, then that could work as well. I don't really have a strong preference, I just wouldn't want the validation to get missed if we end up in an in between state where input-date is being used in a place without whatever wrapper we use to do this automatic validation (not sure if that's a potential case).

Reply:
@dlockhart and I discussed this and we think it makes sense to have components be responsible for calling `validate` themselves so this is the approach we want to take.

The change I would make is to change it to:
```
const errors = await this.validate();
if (errors.length > 0) return;
```
The reason is that `validate` internally calls `checkValidity()` and also handled any `d2l-validation-custom` elements so its always safer to go with `validate`. `checkValidity()` only checks the result of `setValidity(...)`